### ISSUE TITLE
New User/Installation - Can't login to Digikey, kicost hangs and session locks, never completes

### ISSUE THREAD

**Issue by craigarno** [hildogjr/KiCost#522]
#### Issue / Problem report
1. Add the KiCost version. Type ``kicost --version`` at terminal.
2. KiCad\6.0.6>kicost --version
KiCost v1.1.11
KiCAD 6.0.6
Python 3.9.10

3. Add a terminal error report / txt, use ``--debug`` option on KiCost call;
    kicost -w -i %YOUR_FILE% --debug 8

4. Write the command used to call KiCost (or the graphical interface configuration);
kicost -w -i "USB-C PWR.xml" --debug 8

5. One BoM to reproduce the error (with the EDA version).
[555.zip](https://github.com/hildogjr/KiCost/files/9136687/555.zip)
This BOM produces the same error on my setup (made using the same tools) as the "real BOM", which makes me think there is something about my Config.yaml credentials Digikey isn't liking.  I'm not sure why because I can login to the Web Portal using the same credentials, and it works.

Steps;
Running KiCAD/KiCost under Windows 10 Professional from the command line.  I like the GUI, but wasn't able to make that work either.

1. I created a Config.yaml file in ~\.config\kicost, disabling all except Digikey
2. Used my web portal login credentials which work when I visit Digikey's site using a web browser.  I simplified my password to just letters and numbers, like "lqB6U9xo6IwJz81z"
3. When I look up the part with the Web Portal, GRM1555C1H221JA01D is found on Digikey
4. 
![image](https://user-images.githubusercontent.com/4262930/179632776-ff3d20be-e981-44b4-b67d-dab98206ddb9.png)
and then my browser pops up with
https://api.digikey.com/v1/oauth2/authorize?client_id=[ID removed]&response_type=code&redirect_uri=https%3A%2F%2Flocalhost%3A8139%2Fdigikey_callback
![image](https://user-images.githubusercontent.com/4262930/179633029-d9963a15-9ba5-4d17-b622-85822d9b7619.png)
Client ID and password are verified to work with login to the Digikey Web Portal from a web browser.

Debug Level 8 is here:
[kicost-digikey-login-error.txt](https://github.com/hildogjr/KiCost/files/9136232/kicost-digikey-login-error.txt)

Login credentials are replaced with XXXX's just like the example.

- I watched the tantalizing video https://www.youtube.com/watch?v=AeccxROpDfY
- Installed KiCost, which otherwise seems to be installed correctly
- created my config file https://hildogjr.github.io/KiCost/docs/_build/singlehtml/index.html#document-configuration
- Went through and populated my schematic components with a completed field "mfg#" and a valid Manufacturer's PN.
- Created the BOM
- Ran KiCost
- The XLSX file is produced, but only has one part scraped out of 72 unique parts
- The errors are as shown at the start of this message.
- I can share the XML file discretely with a developer, but am hoping this issue is simple enough that it won't be necessary
- I need Digikey and Mouser to work at a minimum, and I'd like to add others (suggestions welcome)

I spent the day trying to figure out what I missed and am now giving up and asking for help.  What did I miss?  Are KiCost credentials different than my Web Portal Login credentials?

Thank you

**Comment by set-soft** [comment]
Hi @craigarno !
The `client_id` in the configuration file is what Digi-Key names `clientId`. This isn't your username, is a value you get when you register for API usage. 

Please visit: https://developer.digikey.com/get_started
You'll get a `clientId` and a `clientSecret`, this is different from your Digi-Key username and password.

**Comment by craigarno** [comment]
This helped, Thank you.

It would also help if the documentation provided some direction for each of the supported sites.
For Digikey I created my Corporation, added the KiCost App, got my App credentials and installed same in my KiCost Config.yaml file.
There were a lot of options and menus I just guessed at, figuring if I asked for everything I'd be covered.
When running KiCost with Digikey I received a popup for what I suspect is a self-signed certificate giving me dire security warnings, which I had to accept before KiCost would run.  I know enough about networking that this part wasn't a big deal.
![image](https://user-images.githubusercontent.com/4262930/179826548-4c5d9733-e0f9-40f9-830d-52e97858f86a.png)
User Documentation describing this process would be helpful.

I'm off to find out why my design caused KiCost to barf at 54% of the way through.  After cleanup, then off to do the same with Mouser, then throw a dart and select a 3rd supplier to round out my collection of distributors.

I have to admit, getting this working will be worth the pain to take over this onerous Supply Chain/BOM task for my next 4 boards.

I am really glad and thankful KiCost exists!
